pkg/core/native/designate.go

```diff
@@ -151,45 +159,77 @@ func (s *Designate) rolesChanged() bool {
 	return rc == nil || rc.(bool)
 }
 
-func oracleHashFromNodes(nodes keys.PublicKeys) util.Uint160 {
+func (s *Designate) hashFromNodes(r Role, nodes keys.PublicKeys) util.Uint160 {
 	if len(nodes) == 0 {
 		return util.Uint160{}
 	}
-	script, _ := smartcontract.CreateMajorityMultiSigRedeemScript(nodes.Copy())
+	var script []byte
+	switch r {
+	case RoleOracle:
+		script, _ = smartcontract.CreateDefaultMultiSigRedeemScript(nodes.Copy())
+	case RoleP2PNotary:
+		return util.Uint160{}
```

Comment:
Not sure it's meaningful.

Reply:
What should we return here? I think it is nice to have all caches unified.

Reply:
Requesting "designated hash" is not really defined for this role, so we can return error (or even panic) for such requests. The other option is to return notary contract hash which actually implements "1 out of N" contract.

Reply:
Fixed, made 1 out of N.

Reply:
Let's try that, though it's not used directly too.